core/krylovTimeEvolver.cpp

```diff
@@ -699,3 +751,125 @@ std::complex<double>* krylovTimeEvolver::expKrylov(double t, std::complex<double
 	return tmpintKernelExp;
 
 }
+
+
+std::string krylovBasicObservable::retName()
+{
+	return obs_name;
+}
+
+obsType krylovBasicObservable::retType()
+{
+	return type;
+}
+
+
+krylovMatrixObservable::krylovMatrixObservable(const std::string& name, matrix* obser) : krylovBasicObservable(name)
+{
+	dim = obser->m;
+	type = MATRIX_TYPE_OBS;
+	if(dim > 0)
+		tmpBlasVec = new std::complex<double>[dim];
+
+	obs = std::make_unique<matrix>(*obser);
+
+}
+
+krylovMatrixObservable::~krylovMatrixObservable()
+{
+	if (dim > 0)
+		delete[] tmpBlasVec;
+}
+
+std::complex<double> krylovMatrixObservable::expectation(std::complex<double>* vec, int len) //requires testing
```

Comment:
Short answer is yes. I only tested trivial examples. Unfortunately pure BLAS functions are tedious to debug. However, I am confident that it should be correct, but to be on the safe side I marked it as "not fully tested". Someone using it will get a compiler warning.
 3d67bed8